src/native/libs/System.Globalization.Native/pal_locale.m

```diff
@@ -151,6 +152,462 @@
 
     return strdup(value);
 }
+
+// invariant character definitions
+#define CHAR_CURRENCY ((char)0x00A4)   // international currency
+#define CHAR_SPACE ((char)0x0020)      // space
+#define CHAR_NBSPACE ((char)0x00A0)    // space
+#define CHAR_DIGIT ((char)0x0023)      // '#'
+#define CHAR_MINUS ((char)0x002D)      // '-'
+#define CHAR_PERCENT ((char)0x0025)    // '%'
+#define CHAR_OPENPAREN ((char)0x0028)  // '('
+#define CHAR_CLOSEPAREN ((char)0x0029) // ')'
+#define CHAR_ZERO ((char)0x0030)       // '0'
+
+/*
+Function:
+NormalizeNumericPattern
+
+Returns a numeric string pattern in a format that we can match against the
+appropriate managed pattern.
+*/
+static char* NormalizeNumericPattern(const char* srcPattern, int isNegative)
+{
+    int iStart = 0;
+    int iEnd = strlen(srcPattern);
+    int32_t iNegativePatternStart = -1;
+
+    for (int i = iStart; i < iEnd; i++)
+    {
+        if (srcPattern[i] == ';')
+        {
+            iNegativePatternStart = i;
+        }
+    }
+
+    if (iNegativePatternStart >= 0)
+    {
+        if (isNegative)
+        {
+            iStart = iNegativePatternStart + 1;
+        }
+        else
+        {
+            iEnd = iNegativePatternStart - 1;
+        }
+    }
+
+    int index = 0;
+    int minusAdded = false;
+    int digitAdded = false;
+    int currencyAdded = false;
+    int spaceAdded = false;
+
+    for (int i = iStart; i <= iEnd; i++)
+    {
+        char ch = srcPattern[i];
+        switch (ch)
+        {
+            case CHAR_MINUS:
+            case CHAR_OPENPAREN:
+            case CHAR_CLOSEPAREN:
+                minusAdded = true;
+                break;
```

Comment:
I can refactor to 
```
if (strchr(srcPattern,CHAR_MINUS) || strchr(srcPattern,CHAR_OPENPAREN) || strchr(srcPattern,CHAR_CLOSEPAREN))
        minusAdded = true;
```
but this will be slower than what is written now.

Reply:
I wonder if we can use [rangeOfCharacterFromSet](https://developer.apple.com/documentation/foundation/nsstring/1411930-rangeofcharacterfromset/) or if thats any faster.
But now that we have the `if (minusAdded) break;` it should be fine.

Why do we consider `'('` and `')'` as minus added?